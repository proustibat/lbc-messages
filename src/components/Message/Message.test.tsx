import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import { Message, MessageProps } from './index';

expect.extend({ toMatchDiffSnapshot });

const props: MessageProps = {
  id: 'uuid',
  title: 'bla bli blou',
  body: 'body content',
};
describe('Message', () => {
  afterEach(cleanup);

  it('should render correctly by default', () => {
    // Given / When
    const { container } = render(<Message {...props} />);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with isPrivate true', () => {
    // Given / When
    const { container } = render(<Message {...props} isPrivate={true} />);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should toggle the expanded body when clicking the title', () => {
    // Given
    const { asFragment, getByTestId } = render(<Message {...props} />);
    const firstRender = asFragment();

    // When
    fireEvent.click(getByTestId('title'));
    const secondRender = asFragment();

    // Then
    expect(firstRender).toMatchDiffSnapshot(secondRender);
  });
});
