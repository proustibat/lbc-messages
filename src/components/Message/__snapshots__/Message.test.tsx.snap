// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render correctly by default 1`] = `
<div>
  <div
    class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="uuid-content"
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      id="uuid-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
        <p
          class="MuiTypography-root heading_fontSize-0.9375rem heading_marginLeft-8 MuiTypography-body1"
          data-testid="title"
        >
          bla bli blou
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="uuid-header"
            id="uuid-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            >
              <p
                class="MuiTypography-root body_fontSize-0.9375rem body_color-rgba(0, 0, 0, 0.54) MuiTypography-body1"
              >
                body content
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message should render correctly with isPrivate true 1`] = `
<div>
  <div
    class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="uuid-content"
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      id="uuid-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
          />
        </svg>
        <p
          class="MuiTypography-root heading_fontSize-0.9375rem heading_marginLeft-8 MuiTypography-body1"
          data-testid="title"
        >
          bla bli blou
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="uuid-header"
            id="uuid-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            >
              <p
                class="MuiTypography-root body_fontSize-0.9375rem body_color-rgba(0, 0, 0, 0.54) MuiTypography-body1"
              >
                body content
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message should toggle the expanded body when clicking the title 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,20 +1,20 @@
  <DocumentFragment>
    <div
-     class=\\"MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded\\"
+     class=\\"MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded\\"
    >
      <div
        aria-controls=\\"uuid-content\\"
        aria-disabled=\\"false\\"
-       aria-expanded=\\"true\\"
-       class=\\"MuiButtonBase-root MuiAccordionSummary-root Mui-expanded\\"
+       aria-expanded=\\"false\\"
+       class=\\"MuiButtonBase-root MuiAccordionSummary-root\\"
        id=\\"uuid-header\\"
        role=\\"button\\"
        tabindex=\\"0\\"
      >
        <div
-         class=\\"MuiAccordionSummary-content Mui-expanded\\"
+         class=\\"MuiAccordionSummary-content\\"
        >
          <svg
            aria-hidden=\\"true\\"
            class=\\"MuiSvgIcon-root\\"
            focusable=\\"false\\"
@@ -32,11 +32,11 @@
          </p>
        </div>
        <div
          aria-disabled=\\"false\\"
          aria-hidden=\\"true\\"
-         class=\\"MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd\\"
+         class=\\"MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd\\"
        >
          <span
            class=\\"MuiIconButton-label\\"
          >
            <svg
@@ -54,12 +54,12 @@
            class=\\"MuiTouchRipple-root\\"
          />
        </div>
      </div>
      <div
-       class=\\"MuiCollapse-container MuiCollapse-entered\\"
-       style=\\"min-height: 0px;\\"
+       class=\\"MuiCollapse-container\\"
+       style=\\"min-height: 0px; height: 0px; transition-duration: 0ms;\\"
      >
        <div
          class=\\"MuiCollapse-wrapper\\"
        >
          <div"
`;
